File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num_list) {
        int answer = 0;
        StringBuilder even = new StringBuilder();
        StringBuilder odd = new StringBuilder();
        
        for(int num : num_list){
            if(num%2 == 0)
                even.append(num);
            else
                odd.append(num);
        }
        int evenNum = Integer.parseInt(even.toString());
        int oddNum = Integer.parseInt(odd.toString());
        
        answer = evenNum + oddNum;
        return answer;
    }
}

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.js
function solution(num_list) {
    var answer = 0;
    
    let a = num_list.filter(n1 => (n1%2 == 1));
    let b = num_list.filter(n2 => (n2%2 == 0));
    
    let evenNum = Number(a.join(''));
    let oddNum = Number(b.join(''));
    
    answer = evenNum + oddNum;
    
    return answer;
}